tests/rptest/tests/cluster_recovery_test.py

```diff
@@ -0,0 +1,133 @@
+        }
+        si_settings = SISettings(
+            test_context,
+            log_segment_size=self.segment_size,
+            fast_uploads=True,
+        )
+        self.s3_bucket = si_settings.cloud_storage_bucket
+        super(ClusterRecoveryTest, self).__init__(test_context=test_context,
+                                                  si_settings=si_settings,
+                                                  extra_rp_conf=extra_rp_conf)
+
+    @cluster(num_nodes=4)
+    def test_basic_controller_snapshot_restore(self):
+        """
+        Tests that recovery of some fixed pieces of controller metadata get
+        restored by cluster recovery.
+        """
+        rpk = RpkTool(self.redpanda)
+        rpk.cluster_config_set("log_segment_size_jitter_percent", 13)
+
+        for t in self.topics:
+            KgoVerifierProducer.oneshot(self.test_context,
+                                        self.redpanda,
+                                        t.name,
+                                        self.message_size,
+                                        100,
+                                        batch_max_bytes=self.message_size * 8,
+                                        timeout_sec=60)
+        quiesce_uploads(self.redpanda, [t.name for t in self.topics],
+                        timeout_sec=60)
+
+        algorithm = "SCRAM-SHA-256"
+        users = dict()
+        users["admin"] = None  # Created by the RedpandaService.
+        for _ in range(3):
+            user = f'user-{random_string(6)}'
+            password = f'user-{random_string(6)}'
+            users[user] = password
+            self.redpanda._admin.create_user(user, password, algorithm)
+            rpk.acl_create_allow_cluster(user, op="describe")
+        rpk.acl_create_allow_cluster("admin", op="describe")
+
+        time.sleep(5)
+
+        self.redpanda.stop()
+        for n in self.redpanda.nodes:
+            self.redpanda.remove_local_data(n)
+        self.redpanda.restart_nodes(self.redpanda.nodes,
+                                    auto_assign_node_id=True,
+                                    omit_seeds_on_idx_one=False)
+        self.redpanda._admin.await_stable_leader("controller",
+                                                 partition=0,
+                                                 namespace='redpanda',
+                                                 timeout_s=60,
+                                                 backoff_s=2)
+
+        self.redpanda._admin.initialize_cluster_recovery()
+
+        def cluster_recovery_complete():
+            return "recovery_stage::complete" in self.redpanda._admin.get_cluster_recovery_status(
+            ).json()["state"]
+
+        wait_until(cluster_recovery_complete, timeout_sec=30, backoff_sec=1)
+
+        assert len(set(
+            rpk.list_topics())) == 3, "Incorrect number of topics restored"
+        log_jitter_restored = rpk.cluster_config_get(
```

Comment:
Add more validation on the properties of the recovery
- RF
- read replicas
- etc

Reply:
To come in follow-up PRs